engine/kv_engine.cpp

```diff
@@ -561,9 +550,28 @@ Status KVEngine::RestoreSkiplistHead(DLRecord* pmem_record,
   if (s == Status::MemoryOverflow) {
     return s;
   }
-  assert(s == Status::NotFound);
-  hash_table_->Insert(hint, entry_ptr, SortedHeaderRecord, skiplist,
+
+  if (s == Status::Ok &&
+      pmem_record->entry.meta.timestamp < entry_ptr->GetIndex()
+                                              .skiplist->header()
+                                              ->record->entry.meta.timestamp) {
+    // TODO(zhichen): free the expired record
```

Comment:
Don't do it here. You have to rebuild the whole Skiplist to see which Record can be freed.

Reply:
latency code。remove it